stub/src/test/java/io/grpc/stub/ClientCallsTest.java

```diff
@@ -155,8 +155,8 @@ public void start(io.grpc.ClientCall.Listener<String> listener, Metadata headers
       ClientCalls.blockingUnaryCall(call, req);
       fail("Should fail");
     } catch (StatusRuntimeException e) {
-      assertSame(status, e.getStatus());
-      assertSame(trailers, e.getTrailers());
+      assertEquals(status, e.getStatus());
```

Comment:
Here and below: I think you accidentally changes `assertSame` to `assertEquals` in unrelated places.

Reply:
No, it was very much on purpose.  I don't think that in any of those places that it is important for it to be the same object rather than them being equal.  Of course, the same objects are always equal, but it would be better to only check for the meaningful comparison.

Reply:
I don't think it's fair to assume whoever wrote this code didn't consider whether they want to do `assertSame` or `assertEquals`. 
For example, on the line below they do `assertEquals(trailers, e.getTrailers())`, which, IMHO makes a lot of sense here - they're checking that what got propagated in `e.getTrailers()` is the same instance of `Metadata`, as was passed into `listener.onClose(status, trailers);`.

Reply:
Okay, reverted the ones that were in previous tests.